inc/question.class.php

```diff
@@ -259,146 +259,25 @@ private function checkBeforeSave($input) {
-            $input['range_min']      = !empty($input['range_min'])
-                                          ? (float) str_replace(',', '.', $input['range_min'])
-                                          : null;
-            $input['range_max']      = !empty($input['range_max'])
-                                          ? (float) str_replace(',', '.', $input['range_max'])
-                                          : null;
-         }
-
-         // LDAP fields validation
-         if ($input['fieldtype'] == 'ldapselect') {
-            // Fields are differents for dropdown lists, so we need to replace these values into the good ones
-            if (isset($input['ldap_auth'])
-               && !empty($input['ldap_auth'])) {
-
-               $config_ldap = new AuthLDAP();
-               $config_ldap->getFromDB($input['ldap_auth']);
-
-               if (!empty($input['ldap_attribute'])) {
-                  $ldap_dropdown = new RuleRightParameter();
-                  $ldap_dropdown->getFromDB($input['ldap_attribute']);
-                  $attribute     = array($ldap_dropdown->fields['value']);
-               } else {
-                  $attribute     = array();
-               }
-
-               // Set specific error handler to catch LDAP errors
-               if (!function_exists('warning_handler')) {
-                  function warning_handler($errno, $errstr, $errfile, $errline, array $errcontext) {
-                     if (0 === error_reporting()) {
-                        return false;
-                     }
-                     throw new ErrorException($errstr, 0, $errno, $errfile, $errline);
-                  }
-               }
-               set_error_handler("warning_handler", E_WARNING);
-
-               try {
-                  $ds            = $config_ldap->connect();
-                  ldap_set_option($ds, LDAP_OPT_PROTOCOL_VERSION, 3);
-                  ldap_control_paged_result($ds, 1);
-                  $sn            = ldap_search($ds, $config_ldap->fields['basedn'], $input['ldap_filter'], $attribute);
-                  $entries       = ldap_get_entries($ds, $sn);
-               } catch (Exception $e) {
-                  Session::addMessageAfterRedirect(__('Cannot recover LDAP informations!', 'formcreator'), false, ERROR);
-               }
-
-               restore_error_handler();
-
-               $input['values'] = json_encode(array(
-                  'ldap_auth'      => $input['ldap_auth'],
-                  'ldap_filter'    => $input['ldap_filter'],
-                  'ldap_attribute' => strtolower($input['ldap_attribute']),
-               ));
-            }
-         }
-      }
-
-      if ($input['fieldtype'] == 'textarea' || $input['fieldtype'] == 'text') {
-         if (isset($input['default_values'])) {
-            $input['default_values'] = addslashes($input['default_values']);
-         }
-      }
+      $fieldType = 'PluginFormcreator' . ucfirst($input['fieldtype']) . 'Field';
```

Comment:
$input['fieldtype'] contains only a subpart of the class name.  Also, the subpart is a type of field in a similar meaning as we can find in <input> HTML tag 

```html
<input type="text" name="lastname">
```

I would keep the current name because I think it is more accurate.